pages/_conferenceCode/contacts/index.vue

```diff
@@ -366,18 +392,100 @@ function isFiltersApplied (_this){
   else return false
 }
 
+async function asyncFind (query){
+  this.isLoadingOrganization = true
+  if (query !== '')
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      FreeText: query,
+      limit   : 25
+    })
+
+  this.isLoadingOrganization = false
+}
+
 async function mounted (){
+  const qs = this.$route.query
+
   this.meetingsOptions = await getMajorMinorMeetings(this)
 
-  this.organizationOptions = await this.$kronosApi.getOrganizations({
-    limit: 5000
-  })
+  // this.organizationOptions = await this.$kronosApi.getOrganizations({
+  //   limit: 25
+  // })
+
+  this.countryList = this.$store.getters['countries/asSelectOptions']
+
+  sortCountryByAsc(this.countryList, this)
+
+  this.organizationTypesOptions = this.$store.getters[
+    'organizations/asSelectOptions'
+  ]
+
+  setQueryStringFilters(this, qs)
+}
+
+function setQueryStringFilters (_this, qs){
+  if (qs.FreeText) _this.searchTagInput = qs.FreeText
+
+  if (qs.Governments)
+    _this.filter.selectedCountry = setSelectedCountry(_this, qs.Governments)
+
+  if (qs.EventUIDs)
+    _this.filter.selectedMeetings = setSelectedMeetings(_this, qs.EventUIDs)
+
+  if (qs.CountryScope)
+    _this.filter.selectedScop = setCountryScope(_this, qs.CountryScope)
+
+  if (qs.EventRegistrationStatus)
+    _this.filter.selectedAttendance = setRegistrationStatus(
+      _this,
+      qs.EventRegistrationStatus
+    )
+  if (qs.OrganizationUIDs)
+    _this.filter.selectedOrganization = setOrganizations(
+      _this,
+      qs.OrganizationUIDs
+    )
+}
+
+function setOrganizations (_this, organizations){
+  const arrayList = isArrayOrObject(organizations)
+
+  return arrayList.map(o =>
+    _this.organizationOptions.find(x => x.OrganizationUID === o)
+  )
+}
 
-  const country = await this.$kronosApi.getCountries()
+function setRegistrationStatus (_this, regStatus){
+  const arrayList = isArrayOrObject(regStatus)
 
-  sortCountryByAsc(country, this)
+  return arrayList.map(a =>
+    _this.attendanceOptions.find(x => x.value === Number(a))
+  )
+}
 
-  this.organizationTypesOptions = await this.$kronosApi.getOrganizationTypes()
+function setCountryScope (_this, countryScope){
+  return _this.scopeOptions.find(x => x.value === countryScope)
+}
+
+function setSelectedCountry (_this, countryCode){
+  const arrayList = isArrayOrObject(countryCode)
+
+  return arrayList.map(o => _this.countryList.find(x => x.Code === o))
+}
+
+function isArrayOrObject (data){
+  let tempArray = []
+
+  if (Array.isArray(data)) tempArray = data
+  else tempArray.push(data)
+
+  return tempArray
+}
```

Comment:
The function name is wrong. when I read `isArrayOrObject` elsewhere in the code I think it:

> it tell me if the parameter is an `object` or an `array`

but no it return the original array and if it's not an array then it put the value in a new array. it does not return not a true/false as expected byt the function name. 

I recommend renaming it to something more relevant: `asArray()` maybe or toArray() `asList()` or `toList()`. you can use lodash flatten to simplify 

```suggestion
function asArray (data){
  return _.flatten([data]);
}
```